src/main/java/org/apache/commons/compress/utils/IOUtils.java

```diff
@@ -275,6 +277,49 @@ public static byte[] readRange(final ReadableByteChannel input, final int len) t
         return output.toByteArray();
     }
 
+    /**
+     * Writes full buffer to channel.
+     *
+     * @param channel
+     *      channel to write to
+     * @param buf
+     *      buffer to write
+     * @throws IOException
+     *      when writing fails
+     */
+    public static void writeFully(SeekableByteChannel channel, ByteBuffer buf) throws IOException {
```

Comment:
`writeFully` -> `write` or is a "partial write" expected default behavior? That would be odd. Flip the arguments such that the source comes first and the destination second. IOW, "Writes from Foo to Bar", which happens to be the phrasing you use in the Javadoc ;-)

Reply:
The methods in `SeekableByteChannel` have `write` semantics to attempt to write and return as soon as the `write` is attempted.  So I wanted to avoid naming conflict with those and be explicit that this writes full buffer or fails.
Other libraries use the same semantics, such as https://hadoop.apache.org/docs/stable/api/org/apache/hadoop/io/IOUtils.html#writeFully-java.nio.channels.FileChannel-java.nio.ByteBuffer-long-

The example also specifies the same order of parameters which in my opinion makes sense - it operates on object which comes as first parameter, not really in a sense of source of destination.  I can amend this but it's inconsistent with other libraries.